### PR TITLE
mol2rdkit_phys_chem: make failure on any error optional

### PR DESCRIPTION
**Pull request by JochenSiegWork**
Add a parameter fails_on_any_error to MolToRDKitPhysChem to get partial results when the computation of a single physchem descriptor fails.

This feature is necessary for data analysis, where we want to examine data even if one of 200 descriptors failed to be calculated.

### CODE REVIEW

File: molpipeline/mol2any/mol2rdkit_phys_chem.py [hunk 1]
```diff
@@ -54,16 +57,31 @@ def __init__(
...
             UUID of the PipelineElement. If None, a new UUID is generated.
         """
-        self._descriptor_list = descriptor_list or DEFAULT_DESCRIPTORS
+        if descriptor_list is not None:
+            for descriptor_name in descriptor_list:
+                if descriptor_name not in RDKIT_DESCRIPTOR_DICT:
+                    raise ValueError(
+                        f"Unknown descriptor function with name: {descriptor_name}"
+                    )
+            self._descriptor_list = descriptor_list
```
Comment (c-w-feldmann) [inline molpipeline/mol2any/mol2rdkit_phys_chem.py:80]: should we make the `descriptor_list` a property?

File: molpipeline/mol2any/mol2rdkit_phys_chem.py [hunk 2]
```diff
@@ -149,8 +176,10 @@ def set_params(self, **parameters: dict[str, Any]) -> Self:
             Self
         """
         parameters_shallow_copy = dict(parameters)
-        descriptor_list = parameters_shallow_copy.pop("descriptor_list", None)
-        if descriptor_list is not None:
-            self._descriptor_list = descriptor_list  # type: ignore
+        params_list = ["descriptor_list", "return_with_errors", "log_exceptions"]
+        for param_name in params_list:
+            if param_name in parameters:
+                setattr(self, f"_{param_name}", parameters[param_name])
```
Comment (c-w-feldmann) [inline molpipeline/mol2any/mol2rdkit_phys_chem.py:182]: here we dont check the descriptor name. Could be solved with a property.

File: tests/test_elements/test_mol2any/test_mol2rdkit_phys_chem.py [hunk 3]
```diff
@@ -297,10 +295,109 @@ def test_descriptor_normalization(self) -> None:
...
+            np.equal(np.isnan(output2).any(axis=1), [True, False, False]).all()
+        )
+
+    def test_unknown_descriptor_name(self) -> None:
+        """Test the handling of unknown descriptor names."""
+
+        self.assertRaises(
+            ValueError,
+            MolToRDKitPhysChem,
+            **{"descriptor_list": ["__NotADescriptor11Name4374737834hggghgddd"]},
```
Comment (c-w-feldmann) [inline tests/test_elements/test_mol2any/test_mol2rdkit_phys_chem.py:380]: Maybe a bit less randomness in the name :D

**Review by c-w-feldmann — CHANGES_REQUESTED** [review]
Minor changes

File: molpipeline/mol2any/mol2rdkit_phys_chem.py (hunk 2 above)
Comment (JochenSiegWork) [reply molpipeline/mol2any/mol2rdkit_phys_chem.py:182]: You're right. I added a setter that validates the user-defined input data.

**Review by c-w-feldmann — APPROVED** [review]

**Pull request merged**